include/boost/redis/detail/multiplexer.hpp

```diff
@@ -116,8 +116,27 @@ class multiplexer {
       std::size_t read_size_;
    };
 
+   // This constructor makes testing easier
+   explicit multiplexer(bool conn_healthy = false);
+
+   // Handle connection health
+   void on_connection_up();  // Might be called in any state
+
+   void on_connection_down();  // Must be called once, with is_connection_healthy() == true
+
+   bool is_connection_healthy() const { return conn_healthy_; }
```

Comment:
This function should check if conn_healthy_ is not older than `config::health_check_interval`.